### PR TITLE
Fix kube config

### LINKED ISSUE #123: kubeconfig flag is not utilized on app install
The `kubeconfig` flag on the `app install` command does not affect the actual kubernetes command execution. The kubernetes command is executed against the default kubeconfig file, not the one passed in via flags.

## Expected Behaviour

Passing the `kubeconfig` flag on the `app install` command should use that kubeconfig file when installing the app.

## Current Behaviour

The `kubeconfig` variable is only written out, but not utilized:
https://github.com/alexellis/k3sup/blob/master/pkg/cmd/openfaas_app.go#L42

## Possible Solution

Since the `kubernetes_exec.go` file is also in the `cmd` package, it's possible for the `kubectl` and `kubectlTask` functions to look for changes on the flag to get that passed in value. Much the same as the above pasted code from one of the apps.

## Steps to Reproduce (for bugs)

1. `k3sup app install openfaas --kubeconfig any/path/to/file`
2. Output will say something like: "Using kubeconfig {the path from above}"
3. Run `kubectl get pods --all-namespaces --kubeconfig /same/path/above`
4. Observe the app was not installed
5. Run `kubectl get pods --all-namespaces` (without the kubeconfig)
6. See the application was installed on the incorrect cluster

## Context

I have several k8s cluster set up in various environments for testing. Swapping the config files, or env var is annoying, and really time consuming if I forget before I run the install command.

## Your Environment

* What OS or type or VM are you using? Where is it hosted?
Linux/ARM
Desktop (microk8s), GCP, DigitalOcean, Raspberry Pi 3+
* Operating System and version (e.g. Linux, Windows, MacOS):
Linux

### PR DESCRIPTION
**Pull request by phumberdroz**
## Description
Removed duplicated code (kubectl, kubectlTask) and made it one function `kubectl` which is able to serve both cases.
## Motivation and Context
fixes #123 for kubectl 
helm3 needs to be implemented possibly in a similar way

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/cmd/apps.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func MakeApps() *cobra.Command {
 		SilenceUsage: true,
 	}
 
-	install.PersistentFlags().String("kubeconfig", "kubeconfig", "Local path for your kubeconfig file")
```
Comment (alexellis) [inline pkg/cmd/apps.go]: Why did you remove the default value?

File: pkg/cmd/kubernetes_exec.go [hunk 2]
```diff
@@ -219,45 +227,17 @@ func helmInit() error {
...
+		parts = append(parts, "--kubeconfig")
+		parts = append(parts, kubeConfigPath)
 	}
-
-	res, err := task.Execute()
-
-	return res, err
-}
-
-func kubectl(parts ...string) error {
```
Comment (alexellis) [inline pkg/cmd/kubernetes_exec.go]: Are you dropping this method completely now?

File: pkg/cmd/certmanager_app.go [hunk 3]
```diff
@@ -71,7 +67,7 @@ func makeInstallCertManager() *cobra.Command {
 			}
 		}
 
-		nsRes, nsErr := kubectlTask("create", "namespace", namespace)
+		nsRes, nsErr := kubectl(kubeConfigPath, "", "create", "namespace", namespace).Execute()
```
Comment (alexellis) [inline pkg/cmd/certmanager_app.go]: What's the empty parameter for?

File: pkg/cmd/apps.go (hunk 1 above)
Comment (phumberdroz) [reply pkg/cmd/apps.go]: The default value was populated into `command.Flags().GetString("kubeconfig")` and not a valid value since it is not a path

File: pkg/cmd/certmanager_app.go (hunk 3 above)
Comment (phumberdroz) [reply pkg/cmd/certmanager_app.go]: Initially I had the idea to also add kube context as a parameter and wanted to have it prepared for it.

The new function signature of kubectl(kubeConfig, kubeContext, ...args) where kubeContext is not yet used.

File: pkg/cmd/kubernetes_exec.go (hunk 2 above)
Comment (phumberdroz) [reply pkg/cmd/kubernetes_exec.go]: yes there is only one method for now which return ExecTask

File: pkg/cmd/apps.go (hunk 1 above)
Comment (phumberdroz) [reply pkg/cmd/apps.go]: I did not set the default value to `~/.kube/config` because of possible cross platform issues and kubectl respecting the ENV variable KUBECONFIG by default so there should be no need to set it via k3sup as `--kubeconfig`

**Comment by phumberdroz** [comment]
@alexellis can   you give me an update on this PR ?

**Comment by burtonr** [comment]
Hey @phumberdroz it looks like this is in need of a rebase now since just yesterday the `cmd` directory got moved up a level

File: pkg/cmd/apps.go (hunk 1 above)
Comment (alexellis) [reply pkg/cmd/apps.go]: 👍

File: pkg/cmd/inletsoperator_app.go [hunk 4]
```diff
@@ -34,11 +34,7 @@ func makeInstallInletsOperator() *cobra.Command {
 	inletsOperator.Flags().String("pro-client-image", "", "Docker image for inlets-pro's client")
 
 	inletsOperator.RunE = func(command *cobra.Command, args []string) error {
-		kubeConfigPath := getDefaultKubeconfig()
-
-		if command.Flags().Changed("kubeconfig") {
-			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
-		}
+		kubeConfigPath, _ := command.Flags().GetString("kubeconfig")
```
Comment (alexellis) [inline pkg/cmd/inletsoperator_app.go]: So this doesn't seem to use `IsChanged()` or the method that tried to ascertain the best KUBECONFIG to use. 

I can't see any notes or detail on the commits, so please could you add a note explaining why that was removed and how it's been tested?

It'd also be good to get all these commits squashed down into one. 👍

Comment (phumberdroz) [reply pkg/cmd/inletsoperator_app.go]: the logic of this was moved inside of the kubectl function this seems to be a left over and needs to be deleted.

File: pkg/cmd/kubernetes_exec.go [hunk 5]
```diff
@@ -227,7 +228,8 @@ func helmInit() error {
 	return nil
 }
 
-func kubectl(kubeConfigPath string, kubeContext string, parts ...string) execute.ExecTask {
+func kubectl(command *cobra.Command, parts ...string) execute.ExecTask {
```
Comment (phumberdroz) [inline pkg/cmd/kubernetes_exec.go]: @alexellis the important change is here can we agree on this? Then I will update the fork with the changes since it seems that I most likely will have do the cleanup again.

**Comment by alexellis** [comment]
@Waterdrips please could you take a look at this PR?

File: pkg/cmd/certmanager_app.go [hunk 6]
```diff
@@ -71,7 +67,7 @@ func makeInstallCertManager() *cobra.Command {
 			}
 		}
 
-		nsRes, nsErr := kubectlTask("create", "namespace", namespace)
+		nsRes, nsErr := kubectl(command, "create", "namespace", namespace).Execute()
```
Comment (Waterdrips) [inline pkg/cmd/certmanager_app.go]: It looks as though we are passing the entire command into this func just to get the kubeconfig flag. Probably more readable to just pass in the flag every time, then people know why it's sent

Comment (phumberdroz) [reply pkg/cmd/certmanager_app.go]: The Idea of passing down the command came to my mind because currently in every app we do this: 
https://github.com/alexellis/k3sup/blob/master/cmd/apps/certmanager_app.go#L29-L35

Which means in order to use the kubectl function we have to repeat 6 - 12 lines in every app install. (12 when we add kubecontext as a flag) Which IMHO does not quite scale well if we add more and more apps and then making a change to the kubectl function every app is affected.

**Comment by alexellis** [comment]
Thanks for your work on this. I think the solution may have got a little bit hard to manage, can you re-iterate what the issue was and we'll try to fix it, ideally in a simpler way?
